chatgpt_wrapper/asyncChatGPT.py

```diff
@@ -0,0 +1,669 @@
+""" AsyncChatGPT class
```

Comment:
This class creates a ton of code duplication, and would make maintenance far worse and more error prone. What is the purpose/advantage of having a completely separate async class?

Even if it is a good idea, it is definitely out of scope for this PR, and should be put in its own PR.

Reply:
The main problem I encountered is using an async generator in sync context. 

Here are what I have found:

I can't get [this](https://stackoverflow.com/questions/63587660/yielding-asyncio-generator-data-back-from-event-loop-possible/63595496#63595496) to run because I can't handle the `loop` argument. If I get a loop from asyncio it says there's currently no loop. If I create one it says the future I'm `awaiting` is in another loop. I really have little experience in async programming, so teach me if possible :smile: 

I also found another simpler solution, but that only returns all of the generated values when it has finished generating, which makes ask_stream meaningless. 

> Even if it is a good idea, it is definitely out of scope for this PR, and should be put in its own PR.

I'm sorry for the PR title. I'm not very familiar with github, pr and regular workflows. I was wondering that a PR can have only one title and whenever I submit new commits they will just add to this PR. I will create new branches for this :D

Reply:
If this problem is solved I would like to preserve the async one and provide a sync wrapper over that.